source/tutorials/learning-journey/packaging-existing-software.md

```diff
@@ -0,0 +1,589 @@
+:::
+
+
+### `buildInputs` and `nativeBuildInputs`
+Run the last command again:
+```console
+$ nix-build -A icat
+this derivation will be built:
+  /nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv
+building '/nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv'...
+unpacking sources
+unpacking source archive /nix/store/rx21f6fgnmxgp1sw0wbqll9wds4xc6v0-source
+source root is source
+patching sources
+configuring
+no configure script, doing nothing
+building
+build flags: SHELL=/nix/store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15/bin/bash
+gcc -c -Wall -pedantic -std=c99 -D_BSD_SOURCE -o icat.o icat.c
+In file included from /nix/store/hkj250rjsvxcbr31fr1v81cv88cdfp4l-glibc-2.37-8-dev/include/bits/libc-header-start.h:33,
+                 from /nix/store/hkj250rjsvxcbr31fr1v81cv88cdfp4l-glibc-2.37-8-dev/include/stdio.h:27,
+                 from icat.c:31:
+/nix/store/hkj250rjsvxcbr31fr1v81cv88cdfp4l-glibc-2.37-8-dev/include/features.h:195:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcpp-Wcpp8;;]
+  195 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
+      |   ^~~~~~~
+icat.c: In function 'main':
+icat.c:319:33: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
+  319 |                                 write(tempfile, &buf, 1);
+      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
+gcc -o icat icat.o -lImlib2
+installing
+install flags: SHELL=/nix/store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15/bin/bash install
+make: *** No rule to make target 'install'.  Stop.
+error: builder for '/nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv' failed with exit code 2;
+       last 10 log lines:
+       >   195 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
+       >       |   ^~~~~~~
+       > icat.c: In function 'main':
+       > icat.c:319:33: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
+       >   319 |                                 write(tempfile, &buf, 1);
+       >       |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
+       > gcc -o icat icat.o -lImlib2
+       > installing
+       > install flags: SHELL=/nix/store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15/bin/bash install
+       > make: *** No rule to make target 'install'.  Stop.
+       For full logs, run 'nix log /nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv'.
```

Comment:
```suggestion
```

We can see the full logs, no need to have the repeat of the last 10 lines here. Same for other errors.